dsa-23au/java-dsa/jpract-app/src/main/java/com/practice/app/mergeSort.java

```diff
@@ -0,0 +1,75 @@
+package main.java.com.practice.app;
+
+public class mergeSort 
+{
+    public void sort(int[] inputArray)
+    {
+		
+		//Instantiating the variables length and midpoint
+		int length = inputArray.length;
+		int midpoint = length / 2;
+		
+		//setting leftSide size equal to the length
+		//up to midpoint
+		//setting rightSide equal to the remaining length
+		int[] leftSide = new int[midpoint];
+		int[] rightSide = new int[length - midpoint];
+		
+		//if the length is equal to 2, it is already
+		//in order, there is no dividing possible
+		//or necessary
+		if (length < 2)
+		{
+			return;
+		}
+		
+		//for inputs greater than that it is splitting them
+		//by filling the leftSide and rightSide
```

Comment:
Rather than filling up two new arrays for the left and right sides of the arrays, try to use the main array but only look at the integers defined. running two loops makes this a lot less efficient.